### PR TITLE
SONAR-6233 Searching for rules without debt characteristic is not working

### PR DESCRIPTION
**Pull request by julienlancelot**
SONAR-6233 Searching for rules without debt characteristic is not working

### CODE REVIEW

File: server/sonar-server/src/test/java/org/sonar/server/rule/index/RuleIndexMediumTest.java [hunk 1]
```diff
@@ -331,13 +323,12 @@ public void search_by_has_debt_characteristic() throws InterruptedException {
...
     dbSession.commit();
 
-    // 0. assert base case
     assertThat(index.search(new RuleQuery(), new QueryContext()).getTotal()).isEqualTo(2);
-    assertThat(db.debtCharacteristicDao().selectCharacteristics()).hasSize(2);
 
-    // 1. assert hasSubChar filter
     assertThat(index.search(new RuleQuery().setHasDebtCharacteristic(true), new QueryContext()).getTotal())
       .isEqualTo(2);
+    assertThat(index.search(new RuleQuery().setHasDebtCharacteristic(false), new QueryContext()).getTotal())
```
Comment (jblievremont) [inline server/sonar-server/src/test/java/org/sonar/server/rule/index/RuleIndexMediumTest.java]: I would add a rule with no characteristic in the fixture to check that the filter works as expected.

Comment (julienlancelot) [inline server/sonar-server/src/test/java/org/sonar/server/rule/index/RuleIndexMediumTest.java]: Indeed !

**Comment by julienlancelot** [comment]
I've made some update, can you have a look please ?

**Comment by jblievremont** [comment]
LGTM

**Pull request closed without merging**